dll/win32/shell32/changenotify.cpp

```diff
@@ -625,21 +659,47 @@ SHChangeNotifyTransmit(LONG lEvent, UINT uFlags, LPCITEMIDLIST pidl1, LPCITEMIDL
 EXTERN_C void WINAPI
 SHChangeNotify(LONG wEventId, UINT uFlags, LPCVOID dwItem1, LPCVOID dwItem2)
 {
-    LPITEMIDLIST pidl1 = NULL, pidl2 = NULL, pidlTemp1 = NULL, pidlTemp2 = NULL;
+    TRACE("(0x%08x,0x%08x,%p,%p)\n", wEventId, uFlags, dwItem1, dwItem2);
+
+    if (!GetNotificationServer(FALSE))
+        return;
+
     DWORD dwTick = GetTickCount();
+
     WCHAR szPath1[MAX_PATH], szPath2[MAX_PATH];
+    LPITEMIDLIST pidl1 = NULL, pidl2 = NULL;
     LPWSTR psz1, psz2;
-    TRACE("(0x%08x,0x%08x,%p,%p)\n", wEventId, uFlags, dwItem1, dwItem2);
+    DWORD_ITEMIDLIST dwidl;
+    DWORD dwType, dwValue;
```

Comment:
Similarly to the iDrive1,2 below, the dwValue can be defined only where it's needed.
```suggestion
    DWORD dwType;
```

Reply:
Fixed.